libs/ui/pages/mappers.ts

```diff
@@ -9,19 +9,35 @@ export const toUpdateProjectDTO = (
     sourceLanguage: string,
     targetLanguages: string[],
 ): UpdateProjectDTO => {
-    const languageMap: any = {};
-    languages.forEach(
-        (language: any) => (languageMap[language.name] = language),
-    );
-    const source: UpsertProjectLanguageDTO = languageMap[sourceLanguage];
-    const target: UpsertProjectLanguageDTO[] = [];
-    targetLanguages.forEach((obj: any) => target.push(languageMap[obj]));
-
+    const source = languages.find(language => language.name === sourceLanguage)
+    const targets = targetLanguages.map(language => languages.find(lang => lang.name === language))
     const res: UpdateProjectDTO = {
         name: projectName,
-        sourceLanguage: source,
-        languages: target,
+        sourceLanguage: toUpsertProjectLanguageDTO(source),
+        languages: toUpsertProjectLanguageDTOArray(targets)
     };
-
     return res;
 };
+
+
+const toUpsertProjectLanguageDTO = (object?: UpsertProjectLanguageDTO): UpsertProjectLanguageDTO => {
+    if (object === undefined) {
+        return { name: '', abbreviation: '' }
+    }
+    return object
+}
```

Comment:
```suggestion
const toUpsertProjectLanguageDTO = (object: UpsertProjectLanguageDTO): UpsertProjectLanguageDTO => {
    return object
}
```

Your object shouldn't be empty, if he is you should stop the submit (you don't want to add a term with an empty language)